### PR TITLE
Initial bot settings

### PR DESCRIPTION
**Pull request by mdrgoncalves**
Changes throughout the repository.

- Set initial bot settings for its full operation.
- Created commands: Hello and FxTweet.

### CODE REVIEW

File: src/Commands.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Command } from './Command';
+import { FxTweet } from './commands/FxTweet';
+import { Hello } from './commands/Hello';
+
+export const Commands: Command[] = [
+    Hello, 
+    FxTweet
+];
```
Comment (lcsmuller) [inline src/Commands.ts]: Nit: type may be inferred
```suggestion
export const Commands = [
    Hello, 
    FxTweet
];
```

File: src/commands/FxTweet.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const modifiedUrl = inputURL.replace(
+      'https://twitter.com',
+      'https://fxtwitter.com',
+    );
+
+    await interaction.reply({
+      content: modifiedUrl,
+    });
+  },
+};
```
Comment (lcsmuller) [inline src/commands/FxTweet.ts]: Nit: the following will expand to more precise typing
```suggestion
export const FxTweet = {
  name: 'fxtweet',
  description: 'Fix a tweet video embed',
  type: ApplicationCommandType.ChatInput,
  options: [
    {
      name: 'tweet',
      description: 'Twitter URL',
      type: ApplicationCommandOptionType.String,
      required: true,
    },
  ],
  run: async (client: Client, interaction: CommandInteraction) => {
    const inputURL = interaction.options.get('tweet')?.value as string;

    if (!inputURL.startsWith('https://twitter.com')) {
      await interaction.reply({
        content: 'Please provide a valid Twitter URL, b-baka!',
        ephemeral: true,
      });
      return;
    }

    const modifiedUrl = inputURL.replace(
      'https://twitter.com',
      'https://fxtwitter.com',
    );

    await interaction.reply({
      content: modifiedUrl,
    });
  },
} as const satisfies Command;
```

File: src/commands/FxTweet.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      });
+      return;
+    }
+
+    const modifiedUrl = inputURL.replace(
+      'https://twitter.com',
+      'https://fxtwitter.com',
+    );
+
+    await interaction.reply({
```
Comment (lcsmuller) [inline src/commands/FxTweet.ts]: Nit: async may be omitted by returning Promise
```suggestion
  run: (client: Client, interaction: CommandInteraction): Promise<void> => {
    const inputURL = interaction.options.get('tweet')?.value as string;

    if (!inputURL.startsWith('https://twitter.com')) {
      return interaction.reply({
        content: 'Please provide a valid Twitter URL, b-baka!',
        ephemeral: true,
      });
    }

    const modifiedUrl = inputURL.replace(
      'https://twitter.com',
      'https://fxtwitter.com',
    );

    return interaction.reply({
```

File: src/commands/Hello.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  run: async (client: Client, interaction: CommandInteraction) => {
+    const username = interaction.user?.username;
+    const content = `Ara ara... Hi, ${username}-kun!`;
+
+    await interaction.reply({
+      ephemeral: true,
+      content,
+    });
+  },
+};
```
Comment (lcsmuller) [inline src/commands/Hello.ts]: Nit: ditto
```suggestion
export const Hello = {
  name: 'hello',
  description: 'Returns a greeting',
  type: ApplicationCommandType.ChatInput,
  run: (client: Client, interaction: CommandInteraction) => {
    const username = interaction.user?.username;
    const content = `Ara ara... Hi, ${username}-kun!`;

    return interaction.reply({
      ephemeral: true,
      content,
    });
  },
} as const satisfies Command;
```

File: src/index.ts [hunk 5]
```diff
@@ -1,20 +1,17 @@
...
-		msg.channel.send('pong!');
-		console.log('pong-ed ' + msg.author.username);
-	}
-});
+ready(client);
+interactionCreate(client);
 
-client.login('BOT_TOKEN');
\ No newline at end of file
+client.login(process.env.BOT_TOKEN);
```
Comment (lcsmuller) [inline src/index.ts:17]: I think this may be fetched from a `.env`

File: src/listeners/interactionCreate.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { Client, CommandInteraction, Interaction } from 'discord.js';
+import { Commands } from '../Commands';
+
+export default (client: Client): void => {
+  client.on('interactionCreate', async (interaction: Interaction) => {
+    if (interaction.isCommand()) {
+      await handleSlashCommand(client, interaction);
+    }
+  });
+};
```
Comment (lcsmuller) [inline src/listeners/interactionCreate.ts:10]: Ditto
```suggestion
export default (client: Client): void => {
  client.on('interactionCreate', (interaction: Interaction) => {
    if (interaction.isCommand()) {
      return handleSlashCommand(client, interaction);
    }
  });
};
```

File: src/listeners/interactionCreate.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  interaction: CommandInteraction,
+): Promise<void> => {
+  const slashCommand = Commands.find(command => command.name === interaction.commandName);
+  if (!slashCommand) {
+    interaction.followUp({ content: 'An error has occurred' });
+    return;
+  }
+
+  slashCommand.run(client, interaction);
+};
```
Comment (lcsmuller) [inline src/listeners/interactionCreate.ts]: No use for async here
```suggestion
const handleSlashCommand = (
  client: Client,
  interaction: CommandInteraction,
): Promise<void> => {
  const slashCommand = Commands.find(command => command.name === interaction.commandName);
  if (!slashCommand) {
    interaction.followUp({ content: 'An error has occurred' });
    return;
  }

  slashCommand.run(client, interaction);
};
```

File: src/listeners/interactionCreate.ts (hunk 6 above)
Comment (Guilherme-SG) [reply src/listeners/interactionCreate.ts:10]: Why did you say a pokemon name? Do you like ditto?

Comment (mdrgoncalves) [reply src/listeners/interactionCreate.ts:10]: I choose to harden with Metapod and lock the code from modifications

File: src/index.ts (hunk 5 above)
Comment (mdrgoncalves) [reply src/index.ts:17]: It's obtained from a .env file

File: src/listeners/interactionCreate.ts (hunk 6 above)
Comment (n8eloy) [reply src/listeners/interactionCreate.ts:10]: I use sonic boom

Comment (Guilherme-SG) [reply src/listeners/interactionCreate.ts:10]: Coming from you is more like, life drain

**Pull request closed without merging**